flutter/test/sentry_flutter_util.dart

```diff
@@ -13,6 +14,18 @@ void testTransport({
   );
 }
 
+void testScopeObserver(
+    {required SentryFlutterOptions options,
+    required bool hasNativeScopeObserver}) {
+  try {
+    options.scopeObservers
+        .firstWhere((element) => element.runtimeType == NativeScopeObserver);
+    expect(true, hasNativeScopeObserver);
+  } catch (_) {
+    expect(false, hasNativeScopeObserver);
+  }
```

Comment:
Do we actually need a try catch?
If it throws, means the test failed, and this is what we expect.

Reply:
We'd have to do the catch on the test side, as having `hasNativeScopeObserver `, false is also a valid thing we want to test and not have the code flow interrupted by an exception. I tried using the non-throwing `firstWhereOrNull`, but it is emitting a warning when importing the needed collection dependency.

Reply:
Not sure I understand this correctly, if it's a boolean, we can assert true or false without catching errors, maybe the solution is to not use `firstWhere` but rather something that returns null instead, so it won't throw?

Reply:
Refctored so we don't rely on the throw.